packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -115,6 +115,31 @@ describe('<DataGrid /> - Filter', () => {
       });
       expect(getColumnValues()).to.deep.equal(['Nike']);
     });
+
+    it('should allow object as value and work with valueGetter', () => {
```

Comment:
Could you replicate this test case for the other operators? Only "contains" is covered. The new test cases can be dynamically generated iterating over a `[operator, value, expected]` array to not flood this file with duplicated code.

Reply:
Would it run fast enough? We might have many combinations to tests. Would it make sense to test the comparator unitarily?

Reply:
It will add some ms to run everything, but there're not too many combinations. We have 3 string operators, so 3 test cases. With `valueGetter` tests it will be just more 3 test cases. It's kinda related to https://github.com/mui-org/material-ui-x/pull/1665#discussion_r632487421. The idea to cover everything is to prevent us from accidentally using it inside a filter operator. 

For another PR: We should add some tests for numeric and date operators. We just have for string and boolean operators.